pycloudlib/openstack/tests/test_instance.py

```diff
@@ -0,0 +1,66 @@
+"""Openstack instance tests."""
+from unittest import mock
+
+from pycloudlib.openstack.instance import OpenstackInstance
+
+SERVER_ADDRESSES = {
+    'no_floating': [
+        {
+            'OS-EXT-IPS-MAC:mac_addr': '11:11:11:11:11:11',
+            'addr': '10.0.0.1',
+            'OS-EXT-IPS:type': 'fixed',
+            'version': 4
+        },
+    ],
+    'includes_floating': [
+        {
+            'OS-EXT-IPS-MAC:mac_addr': '22:22:22:22:22:22',
+            'addr': '10.0.0.2',
+            'OS-EXT-IPS:type': 'fixed',
+            'version': 4
+        },
+        {
+            'OS-EXT-IPS-MAC:mac_addr': '33:33:33:33:33:33',
+            'addr': '10.0.0.3',
+            'OS-EXT-IPS:type': 'floating',
+            'version': 4
+        }
+    ]
+}
+
+# While the actual results from openstack will have a lot more values
+# in the dict, 'floating_ip_address' is all we care about
+NETWORK_IPS = [
+    {'floating_ip_address': '10.0.0.4'},
+    {'floating_ip_address': '10.0.0.3'},
+]
```

Comment:
Another nit, but if the code under test incorrectly did the equivalent of `chain(*[d.values() for d in NETWORK_IPS])`, we wouldn't catch that with this input.